### ISSUE TITLE
Specify a State Machine region

### ISSUE THREAD

**Issue by gpsamson** [horike37/serverless-step-functions#64]
I ran into an issue earlier was that I was trying to deploy my service to the `us-west-1` region and State Machines are only supported in the `us-west-2` region.

It'd be great if we could specify the region where the State Machine should be deployed, but from my understanding of Cloud Formation you can't define resources for multiple regions.

Is it best to create a separate serverless.yaml specifically for my State Machines definitions?

**Comment by gpsamson** [comment]
The issue was that I was trying to deploy to the `us-west-1` region and State Machines are only supported in the `us-west-2` region.

It'd be great if we could specify the region where the State Machine should be deployed, but from my understanding of Cloud Formation you can't define resources for multiple regions.

My best bet is two create a separate serverless.yaml specifically for my State Machines definition.

**Comment by horike37** [comment]
@gpsamson 
Thank you for opening up this issue.
Yes, Serverless core and this plugin deploy via CloudFromation, so all resources is in same region.
TBH, I can't introduce a good solution.. Does anyone have a good idea?

**Comment by gpsamson** [comment]
@horike37 I think the best way is to just have a separate serverless.yml for state functions if you need to use a different region.

**Comment by horike37** [comment]
@gpsamson 
Good idea. The feature that reference cf output might be able to allow you to use other state machines in other regions.
https://serverless.com/framework/docs/providers/aws/guide/variables/#reference-cloudformation-outputs

If we can find the way of this, I'll write README how in the tips section.

**Comment by gpsamson** [comment]
@horike37 I took a look at the docs and attempted to separate the step functions into a separate yaml file but it's just too messy and troublesome.

What if the plugin deployed the step functions to the correct region if the step machine region *does not* match the service region? E.g.; the service region is `us-west-1` and the step function region is `us-west-2` so we would create a separate CloudFormation template in `us-west-2` specifically for the step functions. I assume we could hook into `after:deploy:functions`.

What do you think.

**Comment by horike37** [comment]
@gpsamson
Maybe it is possible to hook into `after:deploy:functions`.
However, if we choice the way, it is hard to implement in setp functions plugin. Because, the plugin must deploy via one CloudFormation. If you deploy through plural ones, would be too complex.
What about will you create the feature as plugin of setpfunctions.

**Comment by gpsamson** [comment]
@horike37 Can you expand a little more on what you mean, please?

**Comment by horike37** [comment]
First, let me comfirm your thoughts. my understanding might be wrong.

Is it correct your plan?
- divide into main serverless.yml that region setting is `us-west-1` and sub serverless.yml that region setting is `us-west-2`. Then  statemachine setting is only included in sub serverless.yml.
- when running `sls deploy`, sub serverless.yml is deployed via `after:deploy:functions` hook after main serverless.yml deployed

**Comment by gpsamson** [comment]
No, instead the step functions configuration has its own region setting in serverless.yml. So the step functions configuration would look like:

```
provider:
  name: aws
  runtime: nodejs6.10
  region: us-west-2
stepFunctions:
  stateMachines:
    MyStateMachine:
      region: us-west-2 # This region must support step functions.
      name: MyStateMachine
      definition:
          Panic:
            Type: Pass
            Result: "Panic, Panic, Panic!."
            End: true
```

When the `after:deploy:functions` is called we create a separate CloudFormation template only for Step Functions. So there is only a single serverless.yml file, but we deploy the Step Functions to `us-west-2` instead of adding to the CloudFormation template for `us-west-1`.

**Comment by horike37** [comment]
@gpsamson 
Umm, my thought is negative about creating a separate CloudFormation template.
Because Serverelss core team has been discussing about CF stack spilitting for solving the 200 resources / outputs limit.

Please see the detail.
https://github.com/serverless/serverless/issues/3411
https://github.com/serverless/serverless/pull/3504

Therefore, I guess we will face many unexpected problems.
In conclusion, what aboud will you implement as a plugin for serverless-step-functions?
